src/canopy/models/data_models.py

```diff
@@ -3,10 +3,15 @@
 from typing import Optional, List, Union, Dict, Literal
 
 from pydantic import BaseModel, Field, validator, Extra
+from typing import TypedDict
 
 Metadata = Dict[str, Union[str, int, float, List[str]]]
 
 
+class SparseVector(TypedDict):
```

Comment:
Any reason why not to import `pinecone-text`'s definition of `SparseVector`, instead of duplicating it?

Reply:
Yep the pinecone-text definition yields to an incorrect parsing on KBEncodedDocChunk. Basically Pydantic casts Pinecone Text's SparseVector incorrectly causing sparse indices to be float instead of integers.(definition of SparseVector = Dict[str, Union[List[int], List[float]]]) Pydantic in that cases parses to the latter (List[float]). 
 
In the new serverless floats are not allowed as indices which raises an exception. In any case we should aim to fix that on pinecone-text because this typing is incorrect now. Better to use typed dict for these purposes. For now I defined in canopy to avoid waiting for a new release.